node/pkg/reporter/reporter.go

```diff
@@ -58,58 +51,75 @@ func NewReporter(ctx context.Context, opts ...ReporterOption) (*Reporter, error)
 	log.Info().Msgf("Reporter ticker starting with interval: %v", r.SubmissionInterval)
 	ticker := time.NewTicker(r.SubmissionInterval)
+	// ticker := time.NewTicker(3 * time.Second)
 
 	for {
 		select {
 		case <-ctx.Done():
 			log.Debug().Str("Player", "Reporter").Msg("context done, stopping reporter")
 			return
 		case <-ticker.C:
-			err := r.report(ctx)
-			if err != nil {
-				log.Error().Str("Player", "Reporter").Err(err).Msg("reporting failed")
-			}
+			go func() {
+				err := r.report(ctx)
+				if err != nil {
+					log.Error().Str("Player", "Reporter").Err(err).Msg("reporting failed")
+				}
+			}()
 		}
 	}
 }
 
 func (r *Reporter) report(ctx context.Context) error {
-	pairs := ""
-	for _, submissionPair := range r.SubmissionPairs {
-		pairs += submissionPair.Name + ","
-	}
-	submissionData, err := request.Request[[]SubmissionData](request.WithEndpoint(r.DalEndpoint+"/latest-data-feeds/"+pairs), request.WithTimeout(10*time.Second), request.WithHeaders(map[string]string{"X-API-Key": r.DalApiKey, "Content-Type": "application/json"}))
-
-	if err != nil {
-		return err
-	}
-
 	var feedHashes [][32]byte
 	var values []*big.Int
 	var timestamps []*big.Int
 	var proofs [][]byte
 
-	for _, data := range submissionData {
-		intValue, valueErr := strconv.ParseInt(data.Value, 10, 64)
-		if valueErr != nil {
-			log.Error().Str("Player", "Reporter").Err(valueErr).Msgf("failed to parse value in data: %v", data.Symbol)
-			continue
-		}
+	feedHashesChan := make(chan [32]byte, len(r.SubmissionPairs))
+	valuesChan := make(chan *big.Int, len(r.SubmissionPairs))
+	timestampsChan := make(chan *big.Int, len(r.SubmissionPairs))
+	proofsChan := make(chan []byte, len(r.SubmissionPairs))
+
+	wg := sync.WaitGroup{}
+
+	for _, pair := range r.SubmissionPairs {
+		wg.Add(1)
+		go func(pair SubmissionPair) {
+			defer wg.Done()
+			if value, ok := r.LatestData.Load(pair.Name); ok {
+				submissionData, err := processDalWsRawData(value)
+				if err != nil {
+					log.Error().Str("Player", "Reporter").Err(err).Msg("failed to process dal ws raw data")
+					return
+				}
+
+				feedHashesChan <- submissionData.FeedHash
+				valuesChan <- big.NewInt(submissionData.Value)
+				timestampsChan <- big.NewInt(submissionData.AggregateTime)
+				proofsChan <- submissionData.Proof
+			} else {
+				log.Error().Str("Player", "Reporter").Msgf("latest data for pair %s not found", pair.Name)
+			}
+		}(pair)
+	}
 
-		timestampValue, timestampErr := strconv.ParseInt(data.AggregateTime, 10, 64)
-		if timestampErr != nil {
-			log.Error().Str("Player", "Reporter").Err(timestampErr).Msgf("failed to parse timestamp in data: %v", data)
-			continue
-		}
+	wg.Wait()
 
-		feedHashes = append(feedHashes, data.FeedHash)
-		values = append(values, big.NewInt(intValue))
-		timestamps = append(timestamps, big.NewInt(timestampValue))
-		proofs = append(proofs, data.Proof)
+	close(feedHashesChan)
+	close(valuesChan)
+	close(timestampsChan)
+	close(proofsChan)
```

Comment:
it's just optional suggestion
how about declaring this like followed just after channel declaration
```
defer close()
...
```